backends-common/redis/src/test/java/org/apache/james/backends/redis/RedisTLSSentinelExtension.java

```diff
@@ -0,0 +1,218 @@
+    static final GenericContainer sentinel2 = redisSentinelSupplier.apply("sentinel2");
+    static final GenericContainer sentinel3 = redisSentinelSupplier.apply("sentinel3");
+
+    private RedisMasterReplicaContainerList redisMasterReplicaContainerList;
+    private RedisSentinelContainerList redisSentinelContainerList;
+    private final Network network;
+
+    public RedisTLSSentinelExtension() {
+        this(Network.newNetwork());
+    }
+
+    public RedisTLSSentinelExtension(Network network) {
+        this.network = network;
+        redis1.withNetwork(network);
+        redis2.withNetwork(network);
+        redis3.withNetwork(network);
+        sentinel1.withNetwork(network);
+        sentinel2.withNetwork(network);
+        sentinel3.withNetwork(network);
+    }
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        redis1.start();
+        redis2.start();
+        redis3.start();
+        sentinel1.start();
+        sentinel2.start();
+        sentinel3.start();
+        redisMasterReplicaContainerList = new RedisMasterReplicaContainerList(List.of(redis1, redis2, redis3));
+        redisSentinelContainerList = new RedisSentinelContainerList(List.of(sentinel1, sentinel2, sentinel3));
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        Runnables.runParallel(
+            sentinel1::stop,
+            sentinel2::stop,
+            sentinel3::stop,
+            redis1::stop,
+            redis2::stop,
+            redis3::stop);
+        network.close();
+    }
+
+    @Override
+    public void beforeEach(ExtensionContext extensionContext) {
+        redisMasterReplicaContainerList.forEach(Throwing.consumer(container -> container.execInContainer("redis-cli", "flushall")));
+    }
+
+    @Override
+    public Module getModule() {
+        return new AbstractModule() {
+            @Provides
+            @Singleton
+            public RedisConfiguration provideRedisConfiguration() {
+                return redisSentinelContainerList.getRedisConfiguration();
+            }
+        };
+    }
+
+    @Override
+    public boolean supportsParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return parameterContext.getParameter().getType() == RedisSentinelCluster.class;
+    }
+
+    @Override
+    public Object resolveParameter(ParameterContext parameterContext, ExtensionContext extensionContext) throws ParameterResolutionException {
+        return new RedisSentinelCluster(redisMasterReplicaContainerList, redisSentinelContainerList);
+    }
+
+    private static String createRedisSentinelURI(List<GenericContainer> containers) {
+        return new StringBuilder().append("rediss-sentinel://123@")
+            .append(containers.stream().map(container -> container.getHost() + ":" + container.getMappedPort(SENTINEL_PORT))
```

Comment:
nitpick: containers.stream().map(..) should be extracted into a variable.

Reply:
sr i don't understand your idea

Reply:
The idea is to have less nesting by extracting this code into a variable `containers.stream().map(container -> container.getHost() + ":" + container.getMappedPort(SENTINEL_PORT)`. 

But it's a nitpick so feel free to ignore it.